cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java

```diff
@@ -469,14 +469,19 @@ public void testLogicalTypes() throws Exception {
     Schema dateType = Schema.nullableOf(Schema.of(Schema.LogicalType.DATE));
     Schema timeMicrosType = Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MICROS));
     Schema timeMillisType = Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MILLIS));
-    Schema timestampMicrosType = Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS));
-    Schema timestampMillisType = Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MILLIS));
+    Schema tsMicrosType = Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS));
+    Schema tsMillisType = Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MILLIS));
 
     Assert.assertEquals(dateType, Schema.parseJson(dateType.toString()));
     Assert.assertEquals(timeMicrosType, Schema.parseJson(timeMicrosType.toString()));
     Assert.assertEquals(timeMillisType, Schema.parseJson(timeMillisType.toString()));
-    Assert.assertEquals(timestampMicrosType, Schema.parseJson(timestampMicrosType.toString()));
-    Assert.assertEquals(timestampMillisType, Schema.parseJson(timestampMillisType.toString()));
+    Assert.assertEquals(tsMicrosType, Schema.parseJson(tsMicrosType.toString()));
+    Assert.assertEquals(tsMillisType, Schema.parseJson(tsMillisType.toString()));
+    Assert.assertEquals("[{\"logicalType\":\"date\",\"type\":\"int\"},\"null\"]", dateType.toString());
```

Comment:
I think it makes more sense for these to be:

```
Assert.assertEquals(dateType, Schema.parseJson(...));
```

since we're more concerned with the deserialization part than the serialization part.

Reply:
If the serialization is right, deserialization is tested here: https://github.com/caskdata/cdap/blob/develop/cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java#L475-L479

Reply:
I understand that deserialization of a serialized schema is tested there. But we normally go from the json string to a Schema, not vice versa. In fact, a better test case would be to serialize an Avro schema with the relevant logical types and deserialize it to a CDAP Schema.

Reply:
Makes sense.